### PR TITLE
FIX: Only stream from one data location at a time.

### PR DESCRIPTION
**Pull request by LaurenSpiegel**
S3C-429
Prevent memory problems by no longer double buffering.

### CODE REVIEW

File: lib/routes/routesUtils.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
-const { readySetStream } = require('ready-set-stream');
```
Comment (GiorgioRegni) [inline lib/routes/routesUtils.js]: so sad...

**Comment by LaurenSpiegel** [comment]
@ironman-machine try

File: lib/utilities/retrieveData.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+function retrieveData(locations, dataRetrievalFn,
+    response, logger, errorHandlerFn) {
+    if (locations.length === 0) {
```
Comment (vrancurel) [inline lib/utilities/retrieveData.js]: When does this condition occurs ?

Comment (LaurenSpiegel) [inline lib/utilities/retrieveData.js]: I'm shifting from the array and calling the function recursively so when all is finished the array length is 0.

**Pull request merged**